pkg/verifier/cosign/trustpolicy.go

```diff
@@ -256,6 +277,26 @@ func validate(config TrustPolicyConfig, verifierName string) error {
 		}
 	}
 
+	// validate keyless configuration
+	if config.Keyless != (KeylessConfig{}) {
+		// validate certificate identity specified
+		if config.Keyless.CertificateIdentity == "" && config.Keyless.CertificateIdentityRegExp == "" {
+			return re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("trust policy %s failed: certificate identity or identity regex pattern is required", config.Name))
+		}
+		// validate certificate OIDC issuer specified
+		if config.Keyless.CertificateOIDCIssuer == "" && config.Keyless.CertificateOIDCIssuerRegExp == "" {
+			return re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("trust policy %s failed: certificate OIDC issuer or issuer regex pattern is required", config.Name))
+		}
+		// validate only expression or value is specified for certificate identity
+		if config.Keyless.CertificateIdentity != "" && config.Keyless.CertificateIdentityRegExp != "" {
+			return re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("trust policy %s failed: only one of certificate identity or identity regex pattern should be specified", config.Name))
+		}
+		// validate only expression or value is specified for certificate OIDC issuer
+		if config.Keyless.CertificateOIDCIssuer != "" && config.Keyless.CertificateOIDCIssuerRegExp != "" {
+			return re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("trust policy %s failed: only one of certificate OIDC issuer or issuer regex pattern should be specified", config.Name))
+		}
```

Comment:
Looks like codecov reports L288,292,296  are not covered, is it possible to cover these error conditions with unit tests?